misc/chrome_extension/panel.js

```diff
@@ -13,138 +16,944 @@
 		let queryType = e.options[e.selectedIndex].value;
-		let query = document.getElementById("queryToAdd").value;
-		let field = document.getElementById("fieldToAdd").value;
+		let query = document.getElementById('queryToAdd').value;
+		let field = document.getElementById('fieldToAdd').value;
 
 		//Get the current json
-		let textArea = document.getElementById("json-config");
-		let currentJson = JSON.parse(textArea.value);
+		let textAreaElement = document.getElementById('json-config');
+		let currentJson = JSON.parse(textAreaElement.value);
 
 		//json to add
 		let toAdd = {};
 		toAdd['type'] = queryType;
 		toAdd['path'] = query;
 
 		//Add to current json
-		if(metaType == "exclude"){
-			currentJson[0]["exclude"].push(toAdd);
+		if (metaType == 'exclude') {
+			currentJson[0]['exclude'].push(toAdd);
 		}
-		else if(metaType == "metadata"){
+		else if (metaType == 'metadata') {
 			currentJson[0]['metadata'][field] = toAdd;
 		}
 
 		//Add back to page
 		currentJson = JSON.stringify(currentJson, null, 2);
-		textArea.value = currentJson;
+		textAreaElement.value = currentJson;
+		buildVisualFromText();
 	}
 
-	//Hides or shows the field option
-	function toggleField(){
-		let e = document.getElementById("queryToAddMeta");
-		let fieldElement = document.getElementById('fieldToAdd');
-		if(e.value == "metadata"){
-			fieldElement.style.display = 'inline';
+
+	/**
+	 * Hides or shows the field input
+	 * 
+	 */
+	function toggleQuickAddFieldInput() {
+		let queryTypeElement = document.getElementById('queryToAddMeta');
+		let fieldInputElement = document.getElementById('fieldToAdd');
+		if (queryTypeElement.value == 'metadata') {
+			fieldInputElement.style.display = 'inline';
 		}
-		else{
-			fieldElement.style.display = 'none';
+		else {
+			fieldInputElement.style.display = 'none';
 		}
 	}
 
-	//Resets the value table back to default
-	function resetValueTable(){
-		document.getElementById('resultTable').innerHTML = "<tr><th>Field</th><th>Value(s)</th></tr>";
+
+	/**
+	 * Resets the value table back to default
+	 * 
+	 */
+	function resetResultTable() {
+		document.getElementById('resultTable').innerHTML = '<tr><th>Field</th><th>Value(s)</th></tr>';
+	}
+
+	/**
+	 * Turn on the mouseover on the webpage
+	 * Deactivated for now
+	 * 
+	 */
+	function mouseAdd() {
+		chrome.runtime.sendMessage({ tabId: chrome.devtools.inspectedWindow.tabId, mouse: '1' });
 	}
 
-	//The init function
-	function init() {
 
-		//Adds the fetch function to the button
-		document.getElementById('fetch').onclick = fetch;
-		document.getElementById('pretty').onclick = pretty;
-		document.getElementById('queryToAddButton').onclick = addToJson;
-		document.getElementById("queryToAddMeta").onchange = toggleField;
+	/**
+	 * Checks if the key exists in the metadata of the json
+	 * 
+	 * @param {string} key - The key
+	 * @returns {boolean} True if the key exists
+	 */
+	function DoesKeyExist(key) {
+		let textAreaElement = document.getElementById('json-config');
```

Comment:
these 3 lines to get the JSON from the config is repeated often. Should have a function `getJsonConfig`.